doc/api/esm.md

```diff
@@ -106,11 +106,21 @@ provides interoperability between them and its original module format,
 
 Node.js has two module systems: [CommonJS][] modules and ECMAScript modules.
 
-Authors can tell Node.js to use the ECMAScript modules loader via the `.mjs`
-file extension, the `package.json` [`"type"`][] field, the [`--input-type`][]
-flag, or the [`--experimental-default-type`][] flag. Outside of those cases,
-Node.js will use the CommonJS module loader. See [Determining module system][]
-for more details.
+Authors can tell Node.js to interpret JavaScript as an ES module via the `.mjs`
+file extension, the `package.json` [`"type"`][] field with a value `"module"`,
+the [`--input-type`][] flag with a value of `"module"`, or the
+[`--experimental-default-type`][] flag with a value of `"module"`. These are
+explicit markers of code being intended to run as an ES module.
+
+Inversely, authors can tell Node.js to interpret JavaScript as CommonJS via the
+`.cjs` file extension, the `package.json` [`"type"`][] field with a value
+`"commonjs"`, the [`--input-type`][] flag with a value of `"commonjs"`, or the
+[`--experimental-default-type`][] flag with a value of `"commonjs"`.
+
+When code lacks explicit markers for either module system, Node.js will attempt
+to run JavaScript as CommonJS; if it errors due to syntax that is only supported
+in ES modules, it will try again to run the code as an ES module. See
+[Determining module system][] for more details.
```

Comment:
```suggestion
to run JavaScript as CommonJS; if it errors due to syntax that is only supported
in ES modules, it will try again to run the code as an ES module. See
[Determining module system][] for more details.
```
this is an implementation detail. won't it be better to write something along the lines of "If node.js determines the code uses ES modules it will run as an ES module"?
this will also allow us to change the detection if needed

Reply:
Well I'm not saying which syntax; just that it errors under CommonJS. Sure we could elide that, but shouldn't users know how this is enabled? Some will certainly try to opt into it, so they'll want to know how. We're locking ourselves into only enabling this via syntax that errors in CommonJS, because anything else would be a breaking change.